### ISSUE TITLE
Performance degeneration when batch size is increased(BYOL)

### ISSUE THREAD

**Issue by sooonwoo** [vturrisi/solo-learn#130]
Hi, thanks for your impressive work! 

During the BYOL experiment on your codes(Imagenet-100, resnet18, 200epochs, same hyper-params as bash_files), it seems that the performance degenerated a lot if the batch size was increased([val acc1] bsz128: 75.5%, bsz256:70.54%). 
Is there anything I need to modify when the batch size is increased? 
I don't think it's because of the learning rate

**Comment by DonkeyShot21** [comment]
Hi, thanks for using solo-learn.

The batch sizes (128, 256) that you mention are for each GPU or total?

**Comment by sooonwoo** [comment]
It means I just used configuration in script file as --batch_size 128/256

**Comment by DonkeyShot21** [comment]
Ok, so you used 2 GPUs which means that your effective batch sizes are 256 and 512. We have never tried batch size 512 so I am not able to say for sure, but this might be related to the number of steps. What I mean is that if you train with larger batch sizes you are taking a lower number of gradient steps per epoch. Therefore, a short training (like 200 epochs on IN100) might be negatively affected by this. OFC with larger batches you will probably converge to a better accuracy but it will take more time to do so. 

One experiment that you could run to validate this hypothesis is the following:
```
--batch_size 128
--accumulate_grad_batches 2
```
In this way, the number of gradient steps will be the same as for bs=256 but the batches will be of size 128.

Also, FYI:
- we tried BYOL on IN100 with `--accumulate_grad_batches 16` (as per the original paper) and it did not perform well
- the learning rate is automatically scaled in the base model
- for larger batches, you might also want to tune LARS setting (`--grad_clip_lars` and `--eta_lars`)

**Comment by sooonwoo** [comment]
Ok, appreciate your considerate comments.

So, --batch_size x num_gpus will be the effective batch_sizes. right?

**Comment by DonkeyShot21** [comment]
Yes!
